packages/nextjs/src/config/webpack.ts

```diff
@@ -214,3 +199,56 @@ function checkWebpackPluginOverrides(userSentryWebpackPluginOptions: Partial<Sen
     );
   }
 }
+
+/**
+ * Determine if this is an entry point into which both `Sentry.init()` code and the release value should be injected
+ *
+ * @param entryPointName The name of the entry point in question
+ * @returns `true` if sentry code should be injected, and `false` otherwise
+ */
+function shouldAddSentryToEntryPoint(entryPointName: string): boolean {
+  return entryPointName === 'pages/_app' || entryPointName.includes('pages/api');
+}
+
+/**
+ * Combine default and user-provided SentryWebpackPlugin options, accounting for whether we're building server files or
+ * client files.
+ *
+ * @param buildContext Nexjs-provided data about the current build
+ * @param userPluginOptions User-provided SentryWebpackPlugin options
+ * @returns Final set of combined options
+ */
+function getWebpackPluginOptions(
+  buildContext: BuildContext,
+  userPluginOptions: Partial<SentryWebpackPluginOptions>,
+): SentryWebpackPluginOptions {
+  const { isServer, dir: projectDir, buildId, dev: isDev } = buildContext;
+
+  const hasSentryProperties = fs.existsSync(path.resolve(projectDir, 'sentry.properties'));
```

Comment:
Can we store in a global const since we only wanna do I/O once? Same applies for `serverInclude` and `clientInclude` constants, but since they are not I/O I'm fine with keeping them in the function as well.

Reply:
To be honest I'd rather not. I think that the benefits of having it in the function - both in terms of immediacy of understanding and encapsulation - outweigh whatever speed improvement we gain by saving a single disk access.

Reply:
I disagree. The variable name is pretty descriptive so it's easy to understand (and we should still maintain clarity because of that).

Reply:
Yeah, I suppose you're right on the variable name. I still like the encapsulation, though. But I realized it's kinda a moot point anyway. Getting this information relies on the `buildContext`, which we don't have when we load the file. So we need to do the fetching of the information here, and while technically we could create a global outside of the function, set it here, and then use that the second time through... it's the difference of literally one disk access we're talking about here. Not worth making the code even a tiny bit more complicated, IMHO.